app/src/main/java/de/mathfactory/mooltifill/time/SendTimeWorker.kt

```diff
@@ -0,0 +1,54 @@
+@file:OptIn(ExperimentalCoroutinesApi::class)
+package de.mathfactory.mooltifill.time
+
+import android.content.Context
+import android.util.Log
+import androidx.work.ExistingWorkPolicy
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import de.mathfactory.mooltifill.AwarenessService
+import de.mathfactory.mooltifill.BleMessageFactory
+import de.mathfactory.mooltifill.MooltipassCommand
+import de.mathfactory.mooltifill.MooltipassMessage
+import de.mathfactory.mooltifill.MooltipassPayload
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.launch
+import java.util.concurrent.TimeUnit
+
+class SendTimeWorker(context: Context, params: WorkerParameters): Worker(context, params) {
+
+    private val SET_DATE_INTERVAL_SECONDS = 4096L //Same in iOS: https://github.com/mooltipass/moolticute/blob/master/src/MPDevice.h
+
+    override fun doWork(): Result {
+        CoroutineScope(Dispatchers.IO).launch {
+            sendTimeToDevice()
+        }
+
+        try {
+            //Schedule the next send time task
```

Comment:
Please use PeriodicWorkRequest instead of repeatedly scheduling one-time work here. See also other comment. Thanks!